pkg/operator/encryption/controllers/condition_controller.go

```diff
@@ -84,6 +84,9 @@ func (c *conditionController) sync(_ context.Context, _ factory.SyncContext) (er
 	encryptedGRs := c.provider.EncryptedGRs()
 	currentConfig, desiredState, foundSecrets, transitioningReason, err := statemachine.GetEncryptionConfigAndState(c.deployer, c.secretClient, c.encryptionSecretSelector, encryptedGRs)
 	if err != nil || len(transitioningReason) > 0 {
+		// do not reset previously set condition
+		// transitioningReason means APIServerRevisionNotConverged
+		cond = nil
```

Comment:
cannot follow here, how it solves the BZ.

What you did here is: don't update conditions when we are progressing.

Reply:
don't overwrite previously set condtion

Reply:
like `EncryptionInProgress`

Reply:
or any other